starboard/shared/starboard/player/dmp_demuxer.cc

```diff
@@ -0,0 +1,202 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/shared/starboard/player/dmp_demuxer.h"
+
+#include <algorithm>
+
+namespace starboard {
+namespace shared {
+namespace starboard {
+namespace player {
+namespace video_dmp {
+
+namespace {
+using shared::starboard::media::AudioStreamInfo;
+}
+
+DmpDemuxer::DmpDemuxer(DmpDemuxer& demuxer, SbMediaType type) {
+  SB_DCHECK(demuxer.HasMediaTrack(type));
+
+  dmp_readers_[type] = demuxer.GetDmpReader(type);
```

Comment:
Since we can get dmp reader from a DmpDemuxer, we should turn this ctor into DmpDemuxer(DmpReader*).  So user can call it like:

auto dmp_reader = dmp_demuxer.GetDmpReader(kSbMediaTypeAudio);
auto audio_demuxer = new DmpDemuxer(dmp_reader);

This also allow user to create a DmpDemuxer from a DmpReader.